app/src/test/java/com/segment/analytics/android/integrations/appsflyer/AppsflyerIntegrationTests.java

```diff
@@ -0,0 +1,271 @@
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import android.app.Application;
+import android.content.Context;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import com.appsflyer.AFInAppEventParameterName;
+import com.appsflyer.AppsFlyerLib;
+import com.segment.analytics.Analytics;
+import com.segment.analytics.Properties;
+import com.segment.analytics.Traits;
+import com.segment.analytics.ValueMap;
+import com.segment.analytics.integrations.IdentifyPayload;
+import com.segment.analytics.integrations.Integration;
+import com.segment.analytics.integrations.Logger;
+import com.segment.analytics.integrations.TrackPayload;
+import static  org.mockito.Mockito.*;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.Map;
+
+@RunWith(AndroidJUnit4.class)
+public class AppsflyerIntegrationTests {
+
+    @Test
+    public void testAppsflyerIntegration_ctor_happyFlow() throws Exception {
+        Context context = mock(Context.class);
+        Logger logger = new Logger("test", Analytics.LogLevel.INFO);
+        AppsFlyerLib appsflyer = mock(AppsFlyerLib.class);
+        String appsflyerDevKey = "appsflyerDevKey";
+        boolean isDebug = logger.logLevel != Analytics.LogLevel.NONE;
+        AppsflyerIntegration appsflyerIntegration = new AppsflyerIntegration(context,logger,appsflyer,appsflyerDevKey);
+        Assert.assertEquals(appsflyerIntegration.isDebug , isDebug);
+        Assert.assertEquals(appsflyerIntegration.appsFlyerDevKey, appsflyerDevKey);
+        Assert.assertEquals(appsflyerIntegration.appsflyer, appsflyer);
+        Assert.assertEquals(appsflyerIntegration.logger, logger);
+        Field field = AppsflyerIntegration.class.getDeclaredField("context");
+        field.setAccessible(true);
+
+        Context contextInappsflyerIntegration = (Context) field.get(appsflyerIntegration);
+
+        Assert.assertEquals(contextInappsflyerIntegration, context);
+//        checking the static clause
+        Assert.assertEquals(appsflyerIntegration.MAPPER.get("revenue"), AFInAppEventParameterName.REVENUE);
+        Assert.assertEquals(appsflyerIntegration.MAPPER.get("currency"), AFInAppEventParameterName.CURRENCY);
+
+        reset(context,appsflyer);
+    }
+
+    @Test
+    public void testAppsflyerIntegration_setManualMode_happyFlow() throws Exception {
+        Assert.assertFalse(AppsflyerIntegration.manualMode);
+        AppsflyerIntegration.setManualMode(true);
+        Assert.assertTrue(AppsflyerIntegration.manualMode);
+        AppsflyerIntegration.setManualMode(false);
+        Assert.assertFalse(AppsflyerIntegration.manualMode);
+    }
+
+    @Test
+    public void testAppsflyerIntegration_startAppsFlyer_happyFlow() throws Exception {
+        MockedStatic<AppsFlyerLib> staticAppsFlyerLib = mockStatic(AppsFlyerLib.class);
+        AppsFlyerLib appsFlyerLib = mock(AppsFlyerLib.class);
+        staticAppsFlyerLib.when(AppsFlyerLib::getInstance).thenReturn(appsFlyerLib);
```

Comment:
Duplicate code, you should avoid writing the same code in multiple tests.   
You can extract it to a new helper method like `mockAppsFlyerLib()`